app/models/game.rb

```diff
@@ -57,4 +57,32 @@ def kings_in_check?(color)
     end
     false
   end
+
+  def check_mate?
+    pieces = remaining_pieces
+    checkmate = true
+    friendly = remaining_pieces(color)
+    friendly_moves = friendly.all_moves
+    friendly_moves.each do |move|
+      move
+      checkmate = false unless kings_in_check?
+      undo_move
+      end
+    end
```

Comment:
The way I'd structure this is


1. Is the game in check? (use `check?`)
2. Can the king make a move that doesn't put it in check? (get all available valid moves, check that those moves don't put the king in check)
3. Can the king/other piece capture the piece that has it in check?